### PR TITLE
String resources, npes, declaration/casting redundancies refactor

### PR DESCRIPTION
**Pull request by gubatron**
- Silenced some trivial NPE warnings
 - Removed redundant casts and initializations, and respective import cleanups
 - Trivial lambda refactors

### CODE REVIEW

**Comment by gubatron** [comment]
The final goal with the string resource stuff is to be able to translate the app easily to multiple languages.

File: app/src/main/java/cash/bchd/android_neutrino/MainActivity.java [hunk 1]
```diff
@@ -369,9 +368,11 @@ private void copyToClipboard(String data) {
         Object clipboardService = getSystemService(CLIPBOARD_SERVICE);
         final ClipboardManager clipboardManager = (ClipboardManager) clipboardService;
         ClipData clipData = ClipData.newPlainText("Source Text", data);
-        clipboardManager.setPrimaryClip(clipData);
-        Snackbar snackbar = Snackbar.make(mCLayout, "Address copied clipboard.", Snackbar.LENGTH_LONG);
-        snackbar.show();
+        if (clipboardManager != null) {
+            clipboardManager.setPrimaryClip(clipData);
+            Snackbar snackbar = Snackbar.make(mCLayout, "Address copied clipboard.", Snackbar.LENGTH_LONG);
```
Comment (zquestz) [inline app/src/main/java/cash/bchd/android_neutrino/MainActivity.java]: Shouldn't this string be in the XML below?

File: app/src/main/java/cash/bchd/android_neutrino/SendActivity.java [hunk 2]
```diff
@@ -180,103 +162,98 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
...
-                        txFee = tx.getFee();
-                        inputVals = tx.getInputValuesList();
+        Button send = findViewById(R.id.sendBtn);
+        send.setOnClickListener(v -> {
+            TextInputLayout layout = findViewById(R.id.addressLayout);
+            try {
+                boolean inputError = false;
+                boolean valid = wallet.validateAddress(address.getText().toString());
+                if (!valid) {
+                    layout.setError("INVALID BITCOIN CASH ADDRESS");
```
Comment (zquestz) [inline app/src/main/java/cash/bchd/android_neutrino/SendActivity.java]: Shouldn't this string be in the XML below?

File: app/src/main/java/cash/bchd/android_neutrino/SendActivity.java [hunk 3]
```diff
@@ -180,103 +162,98 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
...
+                    inputError = true;
+                } else {
+                    layout.setError(null);
+                }
+                Amount toSpend;
+                if (inputAmount.getText().toString().equals("")) {
+                    TextInputEditText etAmount = findViewById(R.id.amountInput);
+                    etAmount.setError("INVALID AMOUNT");
```
Comment (zquestz) [inline app/src/main/java/cash/bchd/android_neutrino/SendActivity.java]: And here.

File: app/src/main/java/cash/bchd/android_neutrino/SendActivity.java [hunk 4]
```diff
@@ -180,103 +162,98 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
...
+                if (showingFiat) {
+                    double fiatAmount = Double.valueOf(inputAmount.getText().toString().replace(",", "."));
+                    toSpend = new Amount(ExchangeRates.getInstance().convertToBCH(fiatAmount, Currency.getInstance(fiatCurrency)));
+                } else {
+                    double bchAmount = Double.valueOf(inputAmount.getText().toString().replace(",", "."));
+                    toSpend = new Amount(bchAmount);
+                }
+                if (toSpend.getSatoshis() == 0 || toSpend.getSatoshis() > balance.getSatoshis()) {
+                    TextInputEditText etAmount = findViewById(R.id.amountInput);
+                    etAmount.setError("INVALID AMOUNT");
```
Comment (zquestz) [inline app/src/main/java/cash/bchd/android_neutrino/SendActivity.java]: Another.

File: app/src/main/java/cash/bchd/android_neutrino/SendActivity.java [hunk 5]
```diff
@@ -180,103 +162,98 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
...
-                    e.printStackTrace();
+                if (memo.getText() != null && !memo.getText().toString().equals("")) {
+                    intent1.putExtra("memo", memo.getText().toString());
+                }
+                if (label != null && !label.equals("")) {
+                    intent1.putExtra("label", label);
                 }
+                startActivity(intent1);
+            } catch (Exception e) {
+                Snackbar snackbar = Snackbar.make(sendLayout, "Error creating transaction.", Snackbar.LENGTH_LONG);
```
Comment (zquestz) [inline app/src/main/java/cash/bchd/android_neutrino/SendActivity.java]: All the Snackbar strings as well?

File: app/src/main/java/cash/bchd/android_neutrino/MainActivity.java (hunk 1 above)
Comment (gubatron) [reply app/src/main/java/cash/bchd/android_neutrino/MainActivity.java]: yes

File: app/src/main/java/cash/bchd/android_neutrino/SendActivity.java (hunk 2 above)
Comment (gubatron) [reply app/src/main/java/cash/bchd/android_neutrino/SendActivity.java]: yes, to all your comments with strings, I just started with a couple.

If merged I can continue.

Comment (zquestz) [reply app/src/main/java/cash/bchd/android_neutrino/SendActivity.java]: Excellent. Thanks for the good start!

**Comment by gubatron** [comment]
All String resources in MainActivity, SendActivity, ReceiveActivity extracted.

Also performed a lot of minor cleanups and a bunch of NPE warnings have been taken care of, along with several others related to missuse of switch statements, boolean simplifications, anonymous classes turned into lambda expressions, same variable assignment, unused variables cleaned up, import cleanups.

Broke fixes into individual commits for easier reviewing / cherry-picking.

File: app/src/main/java/cash/bchd/android_neutrino/ReceiveActivity.java [hunk 6]
```diff
@@ -247,8 +244,6 @@ private void displayQRPopup() {
             int width = point.x;
             int height = point.y;
             int smallerDimension = width < height ? width : height;
-            smallerDimension = smallerDimension;
```
Comment (gubatron) [inline app/src/main/java/cash/bchd/android_neutrino/ReceiveActivity.java:250]: 🤣

**Comment by zquestz** [comment]
This looks great, however will wait for Java folks to chime in. My Java foo needs work. =)

**Review by cpacia — APPROVED** [review]

**Pull request merged**